### PR TITLE
Add to testing guidelines in README

### PR DESCRIPTION
**Pull request by bion**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
[Fixes] #<issue_number>

### CODE REVIEW

**Review by natsid — APPROVED** [review]
Thanks!

File: README.md [hunk 1]
```diff
@@ -130,10 +130,27 @@ For the same resource accessed via JSON API the routes should be under the "/api
...
+
+* Assertions should be on the method's `Result` rather than the rendered HTML
+* Assertions may also be on the database state after the controller method has completed
+* Should not rely heavily on mocking
+
+See [AdminProgramControllerTest.java ](https://github.com/seattle-uat/universal-application-tool/pull/167/files#diff-643f94cff692c6554cd33c8e4c542b9f2bc65b4756bf027a623ce8f203d28677) for a good example of a controller test. See the [Play documentation](https://www.playframework.com/documentation/2.8.x/JavaTest#Unit-testing-controllers) for information on framework-provided testing tools.
+
+#### View tests
+
+`BaseHtmlView` provides a number of HTML tag producing methods, for example [`Tag submitButton(String textContents)`](https://github.com/seattle-uat/universal-application-tool/blob/main/universal-application-tool-0.0.1/app/views/BaseHtmlView.java#L33). These methods tend to be fairly simple, with unit tests that are brittle to small, inconsequential changes. Whether or not to test these types of methods is at the discretion of the implementer and code reviewer(s).
```
Comment (carolinedanzi) [inline README.md:149]: Can we add a link for BaseHtmlView?

**Review by carolinedanzi — APPROVED** [review]
Thanks for doing this!

File: README.md (hunk 1 above)
Comment (bion) [reply README.md]: done

**Pull request merged**